orange/src/main/java/com/hartwig/hmftools/orange/algo/purple/PurpleInterpreter.java

```diff
@@ -68,21 +71,25 @@ public class PurpleInterpreter
     private final List<DriverGene> driverGenes;
     @NotNull
     private final LinxRecord linx;
+    @NotNull
+    private final OrangeRefGenomeVersion refGenomeVersion;
```

Comment:
I would suggest to create the `ChromosomalRearrangementsDeterminer` prior to this class and passing that in as a parameter. That way you don't have to pass around more basic properties such as the ref genome version. Those kind of complexities should be captured by the logic inside the classes that are passed in.